src/wallet/wallet.cpp

```diff
@@ -3342,40 +3342,28 @@ bool CWallet::SelectCoinsGroupedByAddresses(std::vector<CompactTallyItem>& vecTa
     return vecTallyRet.size() > 0;
 }
 
-bool CWallet::SelectPrivateCoins(CAmount nValueMin, CAmount nValueMax, std::vector<CTxIn>& vecTxInRet, CAmount& nValueRet, int nPrivateSendRoundsMin, int nPrivateSendRoundsMax) const
+bool CWallet::SelectDenominatedAmounts(CAmount nValueMax, std::set<CAmount>& setAmountsRet) const
```

Comment:
Answered this in Slack but I'd like to repeat it here so that we had a record if someone would dig back later:
1. we do not need the whole set of (mostly the same) amounts, we just need a set of unique values - this should be enough to pick a denom for the new session in `StartNewQueue`;
2.  `GetDenominationsByAmounts` was looping through amounts in reverse i.e. from smaller to larger which (the order) we get "for free" with `std:set`.